Move HookInterface to \Civi\Core namespace

colemanw opened the pull request:
Overview
----------------------------------------
Places the `HookInterface` in a more central namespace to reflect the fact that it's now used for runtime code (originally it was just for unit tests).

Before
----------------------------------------
Class appears to be only meant for unit tests, but used in core code.
Code comment: `// FIXME: Interface misnomer`

After
----------------------------------------
Copied to core namespace. Old interface kept around for whatever is still using it.

Technical Details
----------------------------------------
This interface doesn't actually *do* anything, it's more like a tag. So I think this is a pretty safe change.

colemanw committed:
Copy \Civi\Test\HookInterface to \Civi\Core namespace

colemanw commented on Civi/Core/Event/EventScanner.php:
```diff
@@ -66,8 +66,8 @@ public static function findListeners($target, $self = NULL): array {
     }
 
     $listenerMap = [];
-    // FIXME: Inteface misnomer
```
@totten this comment was the impetus for this change

seamuslee001 commented:
@colemanw guessing this needs maybe an update https://github.com/totten/civix/blob/master/src/CRM/CivixBundle/Resources/views/Code/test-api.php.php#L6 https://github.com/totten/civix/blob/master/src/CRM/CivixBundle/Resources/views/Code/test-headless.php.php#L12

colemanw commented:
@seamuslee001 no those are test classes. I wasn't targeting those.

totten committed:
Civi\Test\HookInterface is now an alias for Civi\Core\HookInterface

This means that we don't need to maintain two copies of the docblock.

There's a strong expectation that the interfaces work equivalently.

It will be more forgiving if a consumer mixes up the names.

totten commented:
@colemanw Great! This name is much more appropriate.

I think we should phase out the name with a `class_alias()` (see attached commit). If you're OK with that, then we can merge when it's passing.

colemanw commented:
Thanks @totten that makes sense
